toolchest_client/tools/tool.py

```diff
@@ -148,11 +148,13 @@ def _merge_outputs(self, output_file_paths):
         """Merges output files for parallel runs."""
         raise NotImplementedError(f"Merging outputs not enabled for this tool {self.tool_name}")
 
-    def _sanity_output_check(self):
-        """
-        Confirms that – at a very basic level – output was created.
-        Tool subclasses can have more specific implementations.
-        """
+    def _preflight(self):
+        """Generic preflight check. Tools can have more specific implementations."""
+        # Validate Toolchest auth key.
+        validate_key()
+
+    def _postflight(self):
+        """Generic postflight check. Tools can have more specific implementations."""
```

Comment:
Created `preflight` and `postflight` methods, which can be overwritten or extended by specific tool implementations. See above for the Kraken 2 implementation.

This will make better validation easier in the future for other use cases as well.